examples/preboot_example/server/plugin.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema } from '@kbn/config-schema';
+import type { CorePreboot, PrebootPlugin, PluginInitializerContext } from 'src/core/server';
+import fs from 'fs/promises';
+import { errors } from '@elastic/elasticsearch';
+import Boom from '@hapi/boom';
+import type { ConfigType } from './config';
+
+export function getDetailedErrorMessage(error: any): string {
+  if (error instanceof errors.ResponseError) {
+    return JSON.stringify(error.body);
+  }
+
+  if (Boom.isBoom(error)) {
+    return JSON.stringify(error.output.payload);
+  }
+
+  return error.message;
+}
+
+export class PrebootExamplePlugin implements PrebootPlugin {
+  readonly #initializerContext: PluginInitializerContext<ConfigType>;
+  constructor(initializerContext: PluginInitializerContext<ConfigType>) {
+    this.#initializerContext = initializerContext;
+  }
+
+  public setup(core: CorePreboot) {
+    const { skipSetup } = this.#initializerContext.config.get<ConfigType>();
+    core.http.registerRoutes('', (prebootRouter) => {
+      prebootRouter.get(
+        {
+          path: '/api/preboot/state',
+          validate: false,
+          options: { authRequired: false },
```

Comment:
Not necessarily, Core authentication system is designed in such a way that all requests are considered authenticated in case no authentication "middleware" (x-pack security plugin) is registered. That was the case for former OSS distribution, and also the case for `preboot` stage.

I use `authRequired: false` here mostly for expressiveness, but we obviously can omit it in `preboot` plugins.